plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/gramine/GenerateGramineDirectManifest.kt

```diff
@@ -71,23 +62,79 @@ open class GenerateGramineDirectManifest @Inject constructor(
+         * All other enclaves are built in the container.
+         */
+        if (pythonEnclave.get()) {
+            val architecture = commandWithOutput("gcc", "-dumpmachine").trimEnd()
+            val ldPreload = executePython(
+                "from sysconfig import get_config_var; " +
+                        "print(get_config_var('LIBPL') + '/' + get_config_var('LDLIBRARY'))"
+            )
+            // The location displayed by 'pip3 show jep' is actually of the site/dist-packages dir, not the specific 'jep'
+            // dir within it. We assume this is the packages dir for other modules as well. If this assumption is
+            // incorrect then we'll need to come up with a better solution.
+            val pythonPackagesPath = commandWithOutput("pip3", "show", "jep")
+                .splitToSequence("\n")
+                .single { it.startsWith("Location: ") }
+                .substringAfter("Location: ")
+
+            commandLine(
+                listOf(
+                    "gramine-manifest",
+                    "-Djava_home=${System.getProperty("java.home")}",
+                    "-Darch_libdir=/lib/$architecture",
+                    "-Dld_preload=$ldPreload",
+                    "-Dpython_packages_path=$pythonPackagesPath",
+                    "-Dis_python_enclave=${pythonEnclave.get()}",
+                    "-Denclave_mode=${buildType.name.uppercase()}",
+                    "-Dsimulation_mrsigner=${computeSigningKeyMeasurement().toHexString()}",
+                    "-Denclave_worker_threads=$enclaveWorkerThreadCount",
+                    "-Dgramine_max_threads=$gramineMaxThreads",
+                    "-Denclave_size=${if (pythonEnclave.get()) PYTHON_ENCLAVE_SIZE else JAVA_ENCLAVE_SIZE}",
+                    manifestTemplateFile.absolutePathString(),
+                    manifestFile.asFile.get().absolutePath
+                )
             )
-        )
+        } else {
+            // We want to call gcc and python from inside the container.
+            val architecture = linuxExec.execWithOutput(listOf("gcc", "-dumpmachine"))
+            val ldPreload = linuxExec.execWithOutput(
+                listOf(
+                    "python3", "-c",
+                    "from sysconfig import get_config_var; " +
+                            "print(get_config_var('LIBPL') + '/' + get_config_var('LDLIBRARY'))"
+                )
+            )
+            // The location displayed by 'pip3 show jep' is actually of the site/dist-packages dir, not the specific 'jep'
+            // dir within it. We assume this is the packages dir for other modules as well. If this assumption is
+            // incorrect then we'll need to come up with a better solution.
+            val pythonPackagesPath = linuxExec.execWithOutput(listOf("pip3", "show", "jep"))
+                .splitToSequence("\n")
+                .single { it.startsWith("Location: ") }
+                .substringAfter("Location: ")
```

Comment:
These two bits are not relevant in this else case, so set them to empty string.

Reply:
Done. Plus a refactor to avoid duplication.